addons/account/wizard/account_payment_register.py

```diff
@@ -526,31 +528,35 @@ def _compute_partner_bank_id(self):
     @api.depends('payment_type', 'journal_id', 'currency_id')
     def _compute_payment_method_line_fields(self):
         for wizard in self:
-            if wizard.journal_id:
-                wizard.available_payment_method_line_ids = wizard.journal_id._get_available_payment_method_lines(wizard.payment_type)
-            else:
-                wizard.available_payment_method_line_ids = False
+            wizard.available_payment_method_line_ids = wizard._get_available_payment_method_lines(wizard.payment_type)
+
+    def _get_available_payment_method_lines(self, payment_type):
+        for wizard in self:
+            available_payment_method_line_ids = wizard._get_manual_payment_method_line(payment_type)
+            for journal in wizard.available_journal_ids:
+                available_payment_method_line_ids |= journal._get_available_payment_method_lines(payment_type)
+        return available_payment_method_line_ids
 
-    @api.depends('payment_type', 'journal_id')
+    @api.depends('payment_type', 'available_payment_method_line_ids')
     def _compute_payment_method_line_id(self):
         for wizard in self:
-            if wizard.journal_id:
-                available_payment_method_lines = wizard.journal_id._get_available_payment_method_lines(wizard.payment_type)
-            else:
-                available_payment_method_lines = False
+            wizard.payment_method_line_id = wizard._get_payment_method_line(wizard.payment_type)
+
+    def _get_payment_method_line(self, payment_type):
+        for wizard in self:
+            available_payment_method_lines = wizard._get_available_payment_method_lines(payment_type)
 
             if available_payment_method_lines and wizard.payment_method_line_id in available_payment_method_lines:
+                payment_method_line_id = wizard.payment_method_line_id
                 continue
 
             # Select the first available one by default.
-            if available_payment_method_lines:
-                move_payment_method_lines = wizard.line_ids.move_id.preferred_payment_method_line_id
-                if len(move_payment_method_lines) == 1 and move_payment_method_lines.id in available_payment_method_lines.ids:
-                    wizard.payment_method_line_id = move_payment_method_lines
-                else:
-                    wizard.payment_method_line_id = available_payment_method_lines[0]._origin
+            move_payment_method_lines = wizard.line_ids.move_id.preferred_payment_method_line_id
+            if len(move_payment_method_lines) == 1 and move_payment_method_lines.id in available_payment_method_lines.ids:
+                payment_method_line_id = move_payment_method_lines
             else:
-                wizard.payment_method_line_id = False
+                payment_method_line_id = available_payment_method_lines[0]._origin
+            return payment_method_line_id
```

Comment:
The `return` conflicts with the `for wizard in self` and also with the `continue`